### ISSUE TITLE
Implement canonical solution for missing MPI attribute key

### ISSUE THREAD

**Issue by bgeihe** [trixi-framework/libtrixi#55]
Thanks a lot for providing this hot fix! Maybe you can also create an issue that lets us track this such that we cannot forget to fix it once there is a canonical solution available from MPI.jl.

_Originally posted by @sloede in https://github.com/trixi-framework/libtrixi/pull/54#pullrequestreview-1531262855_

**Comment by bgeihe** [comment]
Ad `MPI._finished_loading[]`:
I have some difficulty in keeping track of the relation of  `run_load_time_hooks()` and `run_init_hooks()`:
https://github.com/JuliaParallel/MPI.jl/blob/2a2bd68a6dc53660108bbc16690e4972de6318d5/src/environment.jl#L52

**Comment by sloede** [comment]
You're right. The load time hooks are run when the package is loaded (and which do not require MPI to be initialized to work), while the init hooks are only run once MPI is initialized. I got confused by those two 😅

What is run in `MPI.Init` are the init hooks:
https://github.com/JuliaParallel/MPI.jl/blob/2a2bd68a6dc53660108bbc16690e4972de6318d5/src/environment.jl#L145

Based on the implementation on `run_init_hooks`,
https://github.com/JuliaParallel/MPI.jl/blob/2a2bd68a6dc53660108bbc16690e4972de6318d5/src/environment.jl#L53-L59
what we could do in libtrixi is something like
```julia
if !isempty(MPI.mpi_init_hooks)
    MPI.run_init_hooks()
end
```
maybe?

**Comment by sloede** [comment]
Wasn't this fixed by #65 and can be closed?

**Comment by bgeihe** [comment]
Indeed!
